api/types/types.proto

```diff
@@ -2547,7 +2547,9 @@ message OIDCConnectorSpecV3 {
     // RedirectURL is a URL that will redirect the client's browser
     // back to the identity provider after successful authentication.
     // This should match the URL on the Provider's side.
-    string RedirectURL = 4 [ (gogoproto.jsontag) = "redirect_url" ];
+    //
+    // DELETE IN 11.0.0 in favor of RedirectURLs
+    string RedirectURL = 4 [ (gogoproto.jsontag) = "redirect_url_deprecated,omitempty" ];
```

Comment:
As per other comment: can we deprecate this field explicitly, but leave the old tag in place?

Also, for older clients, we could simply fill this field with the first value of `RedirectURLs`. This wouldn't be optimal (performance-wise), but would maintain backwards compatibility.

Reply:
Yeah, the issue is that the gRPC message will be changed, so when old clients/auth servers send `String RedirectURL`, the new clients/servers will return an error since they expect `wrappers.Strings RedirectURL`. And vice versa.

Reply:
Ah yes, you're right, I mixed it up with json unmarshalling for some reason. In that case I think it makes sense. Another option we have is to just introduce a proper new `RedirectURLs` with tag `redirect_urls` and deprecate the old one. 

So it looks like we have 3 options:

1. Try to make existing `redirect_url` support both strings and arrays of strings which is challenging to do in backwards compatible fashion with GRPC.
2. Intro new OIDC connector `redirect_urls` field that accepts array, and deprecate singular `redirect_url` in backwards compatible fashion (which means we'll likely keep it forever but update the docs to use the `redirect_urls`).
3. Let users set `proxy_redirect_addr` in the proxy config instead, in which case we won't need to touch the OIDC connector spec at all.

@Joerger Is that accurate? Let's check with the product team (@klizhentas @xinding33) on which they'd prefer we implement.

Reply:
Correct. Note that with option 1, after the deprecation cycle is complete in v11.0.0, we can actually apply this backwards_compatiblity approach again to get match the json tags:

Reply:
Correct. Note that with option 1, after the deprecation cycle is complete in v11.0.0, we can actually apply this backwards_compatiblity approach again to get match the json tags:

```
message OIDCConnectorSpecV3 {
    ...
    strings RedirectURL = 4 [ (gogoproto.jsontag) = "redirect_url_deprecated" ];
    ...
    wrappers.Strings RedirectURLs = 14 [ (gogoproto.jsontag) = "redirect_url" ];
}
```
---->
```
message OIDCConnectorSpecV3 {
    ...
    wrappers.Strings RedirectURL = 4 [ (gogoproto.jsontag) = "redirect_url" ];
    ...
    wrappers.Strings RedirectURLs = 14 [ (gogoproto.jsontag) = "redirect_url_deprecated" ];
}
```

Reply:
> 1. Try to make existing `redirect_url` support both strings and arrays of strings which is challenging to do in backwards compatible fashion with GRPC.
> 2. Intro new OIDC connector `redirect_urls` field that accepts array, and deprecate singular `redirect_url` in backwards compatible fashion (which means we'll likely keep it forever but update the docs to use the `redirect_urls`).

I also should mention that the backwards compatibility steps taken for (1) and (2) will be the same - Set `oidc.RedirectURL = oidc.RedirectURLs[0]` when sending to old clients/servers, and read `oidc.RedirectURLs[0] = oidc.RedirectURL` when receiving from old clients/servers.

The only difference is that with (1), the json tags will not match the field names:
```
RedirectURL  `json:"redirect_url_deprecated"`
RedirectURLs `json:"redirect_url"`
```

and with (2), they will match:
```
RedirectURL  `json:"redirect_url"`
RedirectURLs `json:"redirect_urls"`
```